flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointStatsTracker.java

```diff
@@ -155,7 +167,12 @@ public CheckpointStatsSnapshot createSnapshot() {
                                 counts.createSnapshot(),
                                 summary.createSnapshot(),
                                 history.createSnapshot(),
-                                latestRestoredCheckpoint);
+                                jobInitializationMetricsBuilder
+                                        .map(
+                                                JobInitializationMetricsBuilder
+                                                        ::buildRestoredCheckpointStats)
+                                        .orElse(Optional.empty())
+                                        .orElse(null));
```

Comment:
Why 2x orElse?

Reply:
Because
1. builder can be empty
2. if it's not empty,  `buildRestoredCheckpointStats` can also return empty result

This is an equivalent of
```
if (builder.isEmpty()) {
  return null;
}
Optional<Stats> stats = builder.get().buildRestoredCheckpointStats();
return stats.orElse(null);
```

Reply:
Ah I see, makes sense. You could probably rewrite it as flatMap().orElse() though...

Reply:
Ahh, nice!